### PR TITLE
Respect ini session.use_strict_mode

### PR DESCRIPTION
**Pull request by rhertogh**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️/❌Only if Session::openSession() and/or Session::writeSession() are overwritten.
| Tests pass?   | ❌
| Fixed issues  | Redis implementation for https://github.com/yiisoft/yii2/pull/18247

### CODE REVIEW

**Comment by rhertogh** [comment]
@samdark I suggest increasing the minor version to 2.1.0 because the Yii2 version constraint is increased significantly.

File: composer.json [hunk 1]
```diff
@@ -18,12 +18,12 @@
         }
     ],
     "require": {
-        "yiisoft/yii2": "~2.0.16",
+        "yiisoft/yii2": "~2.0.38",
```
Comment (samdark) [inline composer.json:21]:
```suggestion
        "yiisoft/yii2": "~2.0.39",
```

File: composer.json [hunk 2]
```diff
@@ -18,12 +18,12 @@
...
         "ext-openssl": "*"
     },
     "require-dev": {
         "phpunit/phpunit": "<7",
-        "yiisoft/yii2-dev": "~2.0.16"
+        "yiisoft/yii2-dev": "~2.0.38"
```
Comment (samdark) [inline composer.json:26]:
```suggestion
        "yiisoft/yii2-dev": "~2.0.39"
```

**Review by samdark — APPROVED** [review]
Looks good except dependency versions.

**Comment by rhertogh** [comment]
> Looks good except dependency versions.

fixed

**Comment by samdark** [comment]
@rhertogh would you please into tests failure? Thanks.

**Comment by rhertogh** [comment]
> @rhertogh would you please into tests failure? Thanks.

On it (also for the MongoDB PR).

**Comment by samdark** [comment]
👍

**Pull request merged**